### DISCUSSION TITLE
Authorization on Kafka when switching from sync processor methods to async processor methods

### DISCUSSION THREAD

**Discussion by FrankIversen** [LGouellec/streamiz#287] [Q&A] [answered]
Hi

We have been working with streams for some time now. We decided to switch from flatmapvalues to flapmapvaluesasync. Flatmapvalues is working without any issues. Unfortunately the async version fails with Error during initializing internal topics :  "Cluster authorization failed" . As the error message suggests it seems to be an authorization issue. We dont have direct control over the kafka cluster so we cant see logs etc, making debugging a bit difficult. My question is: is there any difference between what kind of authorization rights is needed in kafka for sync vs async processors? Why would the async fail when the sync version works?

**Comment by LGouellec** [comment, marked as answer]
Hi @FrankIversen , 

The async processors doesn't work the same as the synchronous. We have multiple intermediate topics, so if you have an authorization issue, it's probably because your principal doesn't have the correct grant access to create topic.

Have a look to this documentation : https://lgouellec.github.io/kafka-streams-dotnet/async-processing.html#example
